cmake/gfortran_flags.cmake

```diff
@@ -0,0 +1,88 @@
+# ================================
+# Set gfortran compile flags
+# ================================
+message(STATUS "Getting gfortran flags")
+
+# Set flags for all build types
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -std=f2008ts -cpp -ffree-line-length-none -fall-intrinsics")
+
+# ================================
+# Set GFORTRAN flags for a SHARED library
+# ================================
+if(BUILD_SHARED_LIBS)
+  # Add any shared library related stuff here
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -shared -fpic")
```

Comment:
I think it's preferable to use [CMake POSITION_INDEPENDENT_CODE](https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html#prop_tgt:POSITION_INDEPENDENT_CODE)